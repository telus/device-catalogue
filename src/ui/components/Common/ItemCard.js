import React from 'react';
import Heading from '@tds/core-heading';
import Card from '@tds/core-card';
import Box from '@tds/core-box';
import Paragraph from '@tds/core-paragraph';
import PriceLockup from '@tds/core-price-lockup';
import Image from '@tds/core-image';
import HairlineDivider from '@tds/core-hairline-divider';

const ItemCard = ({ brand, name, price, image }) => {
  return (
    <Card fullHeight>
      <Box between={4}>
        <Image
          id="itemCard-img"
          src={image}
          alt={name}
          width={200}
          height={200}
        />
        <Heading level="h3" id="itemCard-brand">{brand}</Heading>
        <Heading level="h4" id="device-card-name">
          {name}
        </Heading>
        <PriceLockup
          size="small"
          topText="Starting at"
          price={price}
          signDirection="left"
          id="device-card-price"
        />
        <HairlineDivider />
        <Paragraph size="small">
          TELUS Easy Payment Applied.
          <br></br>
          Taxes due upfront.
        </Paragraph>
      </Box>
    </Card>
  )
}
export default ItemCard;