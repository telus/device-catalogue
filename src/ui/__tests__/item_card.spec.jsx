import React from 'react';
import renderer from 'react-test-renderer';
import ItemCard from '../components/Common/ItemCard';

describe('Item card', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ItemCard brand={"Apple"} name={"iPhone12"} price={121} image={''} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders when not passed all props', () => {
    const tree = renderer
      .create(<ItemCard brand={"Apple"} name={"iPhone12"} price={121} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});