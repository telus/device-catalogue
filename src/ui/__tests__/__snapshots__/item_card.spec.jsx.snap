// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item card renders correctly 1`] = `
<div
  className="sc-bdfBQB cXMDiG sc-hKgJUU itbWqN sc-hKgJUU itbWqN"
>
  <div
    className="sc-bdfBQB dnrHxh"
  >
    <img
      alt="iPhone12"
      className="Image__StyledImage-b16p20-0 DCldW"
      height={200}
      id="itemCard-img"
      src=""
      width={200}
    />
    <h3
      className="Heading__StyledHeading-mpz92r-0 gslrXd"
      data-testid="heading"
      id="itemCard-brand"
    >
      Apple
    </h3>
    <h4
      className="Heading__StyledHeading-mpz92r-0 jGuBnj"
      data-testid="heading"
      id="device-card-name"
    >
      iPhone12
    </h4>
    <div
      className="sc-bdfBQB IqheP sc-jSgvzq gJTxId sc-jSgvzq gJTxId"
    >
      <div
        className="sc-bdfBQB dtABCB"
      >
        <span
          className="Text__StyledText-sc-1m0rr67-0 htpddo"
          size="small"
        >
          Starting at
        </span>
        <div
          className="PriceLockup__StyledRateTextWrapper-crmhv4-4 kCgybC"
        >
          <div
            className="sc-bdfBQB fKojHK sc-gKseQn hqfjIM sc-gKseQn hqfjIM"
          >
            <div
              className="sc-bdfBQB fKojHK"
            >
              <span
                aria-hidden="false"
                className="PriceLockup__StyledDollarSign-crmhv4-2 iteKpH"
                data-testid="dollarSign"
                size="small"
              >
                $
              </span>
              <span
                aria-hidden="false"
                className="PriceLockup__StyledPriceValue-crmhv4-1 kbdTWc"
                data-testid="priceValue"
                size="small"
              >
                121
              </span>
              <span
                className="PriceLockup__StyledPriceValue-crmhv4-1 kbdTWc"
                data-testid="priceValue"
                size="small"
              >
                <span
                  className="Text__StyledText-sc-1m0rr67-0 sc-iBPTik eCTkyE qFagl"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      className="HairlineDivider__StyledHairlineDivider-biamlk-0 lnWyrV"
    />
    <p
      className="Paragraph__StyledParagraph-efl81j-0 dbIdbC"
      size="small"
    >
      TELUS Easy Payment Applied.
      <br />
      Taxes due upfront.
    </p>
  </div>
</div>
`;

exports[`Item card renders when not passed all props 1`] = `
<div
  className="sc-bdfBQB cXMDiG sc-hKgJUU itbWqN sc-hKgJUU itbWqN"
>
  <div
    className="sc-bdfBQB dnrHxh"
  >
    <img
      alt="iPhone12"
      className="Image__StyledImage-b16p20-0 DCldW"
      height={200}
      id="itemCard-img"
      width={200}
    />
    <h3
      className="Heading__StyledHeading-mpz92r-0 gslrXd"
      data-testid="heading"
      id="itemCard-brand"
    >
      Apple
    </h3>
    <h4
      className="Heading__StyledHeading-mpz92r-0 jGuBnj"
      data-testid="heading"
      id="device-card-name"
    >
      iPhone12
    </h4>
    <div
      className="sc-bdfBQB IqheP sc-jSgvzq gJTxId sc-jSgvzq gJTxId"
    >
      <div
        className="sc-bdfBQB dtABCB"
      >
        <span
          className="Text__StyledText-sc-1m0rr67-0 htpddo"
          size="small"
        >
          Starting at
        </span>
        <div
          className="PriceLockup__StyledRateTextWrapper-crmhv4-4 kCgybC"
        >
          <div
            className="sc-bdfBQB fKojHK sc-gKseQn hqfjIM sc-gKseQn hqfjIM"
          >
            <div
              className="sc-bdfBQB fKojHK"
            >
              <span
                aria-hidden="false"
                className="PriceLockup__StyledDollarSign-crmhv4-2 iteKpH"
                data-testid="dollarSign"
                size="small"
              >
                $
              </span>
              <span
                aria-hidden="false"
                className="PriceLockup__StyledPriceValue-crmhv4-1 kbdTWc"
                data-testid="priceValue"
                size="small"
              >
                121
              </span>
              <span
                className="PriceLockup__StyledPriceValue-crmhv4-1 kbdTWc"
                data-testid="priceValue"
                size="small"
              >
                <span
                  className="Text__StyledText-sc-1m0rr67-0 sc-iBPTik eCTkyE qFagl"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      className="HairlineDivider__StyledHairlineDivider-biamlk-0 lnWyrV"
    />
    <p
      className="Paragraph__StyledParagraph-efl81j-0 dbIdbC"
      size="small"
    >
      TELUS Easy Payment Applied.
      <br />
      Taxes due upfront.
    </p>
  </div>
</div>
`;
